spec/controllers/api/teams_controller_spec.rb

```diff
@@ -5,82 +5,156 @@
 describe Api::TeamsController do
   include ControllerHelpers
 
-  let(:bob) { users(:bob) }
+  let!(:bob) { users(:bob) }
   let(:alice) { users(:alice) }
+  let(:team1) { teams(:team1) }
+  let(:team2) { teams(:team2) }
+  let!(:team3) { Fabricate(:non_private_team) }
+  let!(:team4) { Fabricate(:non_private_team) }
+  let(:bobs_private_key) { bob.decrypt_private_key('password') }
+  let!(:team3_user) { team3.teammembers.first.user }
+
   let(:nested_models) { %w[folder account] }
 
   context 'GET index' do
     it 'should get team for search term' do
-      login_as(:alice)
-      get :index, params: { 'q': 'team' }, xhr: true
+      login_as(:bob)
+      get :index, params: { 'q': '2' }, xhr: true
 
       result_json = data.first
 
-      team = teams(:team1)
-
-      expect(result_json['attributes']['name']).to eq team.name
-      expect(result_json['id']).to eq team.id.to_s
+      expect(result_json['attributes']['name']).to eq team2.name
+      expect(result_json['id']).to eq team2.id.to_s
     end
 
     it 'should get all teams for no query' do
-      login_as(:alice)
+      login_as(:bob)
+
       get :index, params: { 'q': '' }, xhr: true
 
+      expect(data.size).to eq(2)
+
       result_json = data.first
 
-      team = teams(:team1)
+      expect(result_json['attributes']['name']).to eq team1.name
+      expect(result_json['id']).to eq team1.id.to_s
 
-      expect(result_json['attributes']['name']).to eq team.name
-      expect(result_json['id']).to eq team.id.to_s
+      result_json = data.second
+
+      expect(result_json['attributes']['name']).to eq team2.name
+      expect(result_json['id']).to eq team2.id.to_s
     end
 
     it 'should get a single team if one team_id is given' do
-      login_as(:alice)
-
-      team = teams(:team1)
+      login_as(:bob)
 
-      get :index, params: { 'team_ids': [team.id] }, xhr: true
+      get :index, params: { 'team_id': team1.id }, xhr: true
 
-      expect(data.count).to eq(1)
+      expect(data).not_to be(Array)
```

Comment:
as far as I know, it still should be an Array, with just one entry !
the UI is just always displaying what the backend is sending back.